web/postcss/roots-merger.ts

```diff
@@ -0,0 +1,50 @@
+// PostCSS plugin for ../script/clean-css.ts to merge all rules
+// with :root selector into the single rule in the beginning of the file
+
+import { rule } from 'postcss'
+import type { ChildNode, Plugin, Rule } from 'postcss'
+
+export const rootsMerger: Plugin = {
+  postcssPlugin: 'roots-merger',
+  prepare() {
+    let rootNodes = new Map<string, ChildNode>()
+    let rulesToRemove: Rule[] = []
+
+    return {
+      OnceExit(root) {
+        rulesToRemove.forEach(cssRule => cssRule.remove())
+
+        let rootRule = rule({ selector: ':root' })
+
+        rootRule.append(...rootNodes.values())
+        if (rootRule.nodes.length > 0) {
+          root.prepend(rootRule)
+        }
+      },
+      Rule(cssRule) {
```

Comment:
We are in the context of PostCSS plugin, let’s use just `rule` since everything here is about `css`

Reply:
It's because we already have file-wide `rule` and I think it's a bad idea to rename it during the import:
`import { rule } from 'postcss'`

Reply:
Using `import { rule } from 'postcss'` is prohibited in PostCSS 8, since your `import`’s PostCSS could be a different version (and AST will contain nodes from different PostCSS versions). It is not a big problem here, but a big problem for plugins published at npm.

The new way is `Rule(rule, { Rule })` — second argument has all classes to build AST.

Reply:
Aha, got it. ✅ fixed